app/Services/Model/Imp/SummitOrderService.php

```diff
@@ -992,6 +1148,139 @@ public function undo()
+    private $payload;
+
+    /**
+     * @var IMemberRepository
+     */
+    private $member_repository;
+
+    /**
+     * @var ISummitAttendeeRepository
+     */
+    private $attendee_repository;
+
+    /**
+     * @var ISummitTicketTypeRepository
+     */
+    private $ticket_type_repository;
+
+    /**
+     * @var ILockManagerService
+     */
+    private $lock_service;
+
+    /**
+     * AutoAssignPrePaidTicketTask constructor.
+     * @param Member|null $owner
+     * @param Summit $summit
+     * @param array $payload
+     * @param IMemberRepository $member_repository
+     * @param ISummitAttendeeRepository $attendee_repository
+     * @param ISummitTicketTypeRepository $ticket_type_repository
+     * @param ITransactionService $tx_service
+     * @param ILockManagerService $lock_service
+     */
+    public function __construct
+    (
+        ?Member $owner,
+        Summit $summit,
+        array $payload,
+        IMemberRepository $member_repository,
+        ISummitAttendeeRepository $attendee_repository,
+        ISummitTicketTypeRepository $ticket_type_repository,
+        ITransactionService $tx_service,
+        ILockManagerService $lock_service
+    )
+    {
+        $this->tx_service = $tx_service;
+        $this->lock_service = $lock_service;
+        $this->owner = $owner;
+        $this->summit = $summit;
+        $this->payload = $payload;
+        $this->member_repository = $member_repository;
+        $this->attendee_repository = $attendee_repository;
+        $this->ticket_type_repository = $ticket_type_repository;
+    }
+
+    public function run(array $formerState): array
+    {
+        $this->formerState = $formerState;
+
+        return $this->tx_service->transaction(function () {
+            $tickets = $this->payload['tickets'];
+            $ticket_dto = $tickets[0];
+
+            $promo_code_val = $ticket_dto['promo_code'] ?? null;
+            if (empty($promo_code_val)) throw new ValidationException("Promo code is required.");
+
+            $type_id = $ticket_dto['type_id'];
+            $order = $this->lock_service->lock('ticket_type.' . $type_id . 'promo_code.' . $promo_code_val .'.sell.lock',
+                function () use ($promo_code_val, $type_id) {
+                    $promo_code = $this->summit->getPromoCodeByCode($promo_code_val);
+                    if (is_null($promo_code)) throw new EntityNotFoundException("Promo code is not found.");
+
+                    $ticket_type = $this->summit->getTicketTypeById($type_id);
+                    if (is_null($ticket_type)) throw new EntityNotFoundException("Ticket Type is not found.");
+
+                    $ticket = $promo_code->getNextAvailableTicketPerType($ticket_type);
+                    if (is_null($ticket))
+                        throw new ValidationException(sprintf("No more available PrePaid Tickets for Promo Code %s", $promo_code->getCode()));
+
+                    $current_user = ResourceServerContext::getCurrentUser(false);
```

Comment:
this is no needed , $owner is already current user

Reply:
done